usr.sbin/devinfo/devinfo.c

```diff
@@ -70,8 +75,38 @@ print_indent(int n)
 
 	if (n < 1)
 		return;
-	asprintf(&s, "%*s", n, " ");
-	printf("%s", s);
+	asprintf(&s, "{P:%*s}", n, " ");
+	xo_emit(s);
+}
+
+/*
+ * Takes a list of key-value pairs in the form
+ * "key1=val1 key2=val2 ..." and prints them according
+ * to xo formatting.
+ */
+static void
+print_kvlist(char *s)
+{
+	char *kv;
+	char *copy = strdup(s);
```

Comment:
`strdup()` just a wrapper over `malloc()` and `memcpy()`, so it should be checked if it returns NULL

Reply:
This good?

```
	char *copy = strdup(s);

	if (copy == NULL) {
		xo_err(1, "No memory!");
		exit(1);
	}
```

Reply:
Yeah (with bsd style)
```
char *copy;
if ((copy = strdup(s)) == NULL)
        xo_err(1, "No memory!");
```

Reply:
Fixed